src/components/atoms/UserAvatar/UserAvatar.tsx

```diff
@@ -97,7 +112,7 @@ export const _UserAvatar: React.FC<UserAvatarProps> = ({
       {showNametag && <div className={nametagClasses}>{user?.partyName}</div>}
       <img
         className={imageClasses}
-        src={avatarSrc}
+        src={userAvatar}
```

Comment:
The way I see it, you can just add the `useImage` value here, and get rid of the `useEffect` and `useState`. It has a fallback src of `DEFAULT_PROFILE_IMAGE`
```suggestion
        src={pictureUrl}
```

Reply:
@denisdimitrov @mike-lvov `useImage` returns fallback img with a delay ( that also depends on the amount of visitors ) after I press "See more" to expand the list of user avatars, so I had to use internal state to have a fallback ready before the load happens and then updating the sate with new img value if we load it successfully
I agree that the stateless version is smoother, but it just doesn't work the way we need it to

Reply:
@alexey-petrov Have you tried alternative approach of encapsulating the logic into a new component, an `<Image src=... fallback=...>` one to replace `<img src=...>`?

Reply:
You should add some documentation about this and why it's written in the way it is. Otherwise someone like me, might change or remove this tomorrow because it looks weird.

Reply:
fixed in 0d6d78b4fd9051c1d8004354d1f29b70c1ba8171 and 8ac293678fe368d40a8ca067a62b6f5d5c178d3a